server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -135,13 +140,37 @@ public static long sizeOfRelocatingShards(
             if (reservedSpace.containsShardId(routing.shardId())) {
                 continue;
             }
-
             final String actualPath = clusterInfo.getDataPath(routing);
             // if we don't yet know the actual path of the incoming shard then conservatively assume it's going to the path with the least
             // free space
             if (actualPath == null || actualPath.equals(dataPath)) {
-                totalSize += getExpectedShardSize(routing, 0L, clusterInfo, null, metadata, routingTable);
+                totalSize += getExpectedShardSize(
+                    routing,
+                    Math.max(routing.getExpectedShardSize(), 0L),
+                    clusterInfo,
+                    null,
+                    metadata,
+                    routingTable
+                );
+            }
+        }
+
+        // Count the STARTED searchable snapshot shards which are unaccounted in the cluster
+        for (ShardRouting shard : node.shardsWithState(ShardRoutingState.STARTED)) {
+            IndexMetadata indexMetadata = metadata.index(shard.index());
+            if (indexMetadata == null) {
+                continue;
+            }
+            if (indexMetadata.isSearchableSnapshot() == false) {
+                continue;
+            }
+            if (reservedSpace.containsShardId(shard.shardId())) {
+                continue;
+            }
+            if (clusterInfo.getShardSize(shard) != null) {
+                continue;
             }
```

Comment:
Can we make this a single if with `&&` conditions instead? It reads slightly verbose and backwards now. (I do know the one above is similar, but anyway I would prefer that for this new new block of code).